gapic/templates/examples/feature_fragments.j2

```diff
@@ -0,0 +1,127 @@
+{#
+ # Copyright (C) 2019  Google LLC
+ #
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ #     http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+ #}
+
+{#
+A careful reader may comment that there is duplication of effort
+between the python verification step and the dispatch/rendering here.
+There is a little, but not enough for it to be important because
+1) Other python artefacts (client libraries, unit tests, and so forth)
+   are generated using templates, so doing the same for generated samples is consistent.
+2) Using jinja for anything requiring real logic or data structures is a bad idea.
+#}
+
+{# response handling macros #}
+
+{% macro handlePrint(elts) %}
+  {# First elment is a format string, remaining elements are the format string parameters #}
+  {# Validating that the number of format params equals #}
+  {# the number of remaining params is handled by real python code #}
+  {% with fmtStr = ('\"' + elts[0] + '\"') |replace("%s", "{}") %}
+print({{ ([fmtStr] + elts[1:])|join(', ') }})
+  {% endwith -%}
+{% endmacro %}
+
+{% macro handleComment(elts) %}
+  {# First elment is a format string, remaining elements are the format string parameters #}
+  {# Validating that the number of format params equals #}
+  {# the number of remaining params is handled by real python code #}
+  {% with fmtStr = elts[0] %}
+# {{ fmtStr|format(*elts[1:]) }}
+  {% endwith %}
+{% endmacro %}
+
+{% macro handleDefine(statement) %}
+{# Python code already verified the form, no need to check #}
+{% set args = statement.split("=") %}
```

Comment:
This could just be

```jinja
{% set lvalue, rvalue = statement.split('=') %}
```

It could also be done with `with` rather than `set`, which might be preferable as I am unsure what impact macros have outside their scope.

Reply:
If jinja variable definitions are global by default even in macros, I will be very sad but not entirely surprised. In any other system, macros don't have lexical scope because they're just pasted in. Will change to using {% with %}